src/main/java/io/supertokens/emailpassword/PasswordHashingUtils.java

```diff
@@ -101,13 +122,139 @@ public static boolean isInputHashInArgon2Format(String hash) {
         return (hash.startsWith("$argon2id") || hash.startsWith("$argon2i") || hash.startsWith("$argon2d"));
     }
 
-//    private static  boolean doesPasswordHashHaveScrpytPrefix(String passwordHash){
-//        // TODO:
-//        return false;
-//    }
+    public static boolean isInputHashInFirebaseSCryptFormat(String hash) {
+        // the firebase scrypt hash we store is a "$" separated string containing an algorithm identifier prefix,
+        // password hash, salt, memory cost, rounds, salt separator
+        // eg. f_scrypt$passwordHash$salt$m=memorycost$r=rounds$s=saltSeparator
+
+        if (doesPasswordHashHaveFireBaseSCryptPrefix(hash)) {
+            String[] separatedHash = hash.split(FIREBASE_SCRYPT_SEPARATOR);
+            if (separatedHash.length != 6) {
+                return false;
+            }
+
+            boolean containsMemCost = false;
+            boolean containsRounds = false;
+            boolean containsSaltSeparator = false;
+
+            for (int i = 3; i < separatedHash.length; i++) {
+                // does hash contain memory cost
+                if (separatedHash[i].startsWith(FIREBASE_SCRYPT_MEM_COST_SEPARATOR)) {
+                    // check that memory cost is a number
+                    try {
+                        Integer.parseInt(separatedHash[3].split(FIREBASE_SCRYPT_MEM_COST_SEPARATOR)[1]);
+                        containsMemCost = true;
+                    } catch (NumberFormatException e) {
+                        return false;
+                    }
+                }
+
+                // does hash contain rounds
+                if (separatedHash[i].startsWith(FIREBASE_SCRYPT_ROUNDS_SEPARATOR)) {
+                    // check that rounds is a number
+                    try {
+                        Integer.parseInt(separatedHash[4].split(FIREBASE_SCRYPT_ROUNDS_SEPARATOR)[1]);
+                        containsRounds = true;
+                    } catch (NumberFormatException e) {
+                        return false;
```

Comment:
no need to return false here cause you won't be setting `containsRounds = true;` anyway, so it will fail the condition after the for loop.